pre_commit_hooks/end_of_file_fixer.py

```diff
@@ -15,13 +19,22 @@ def fix_file(file_obj: IO[bytes]) -> int:
         return 0
     last_character = file_obj.read(1)
     # last_character will be '' for an empty file
-    if last_character not in {b'\n', b'\r'} and last_character != b'':
+    if last_character not in {LF, CR} and last_character != b'':
+        # Look at first line to determine line ending
+        file_obj.seek(0, os.SEEK_SET)
+        first_line = file_obj.readline()
```

Comment:
this will read the whole file into memory for `\r`-delimited files -- and will read the whole file into memory if it has no newlines